packages/swap/src/utils/statechain.ts

```diff
@@ -12,11 +13,67 @@ export type SwapRateArgs = {
   destAsset: InternalAsset;
   amount: bigint;
   limitOrders?: LimitOrders;
+  brokerCommissionBps?: number;
 };
 
 export type LimitOrders = RpcParams['cf_swap_rate_v2'][3];
 
-export const getSwapRateV2 = async ({ srcAsset, destAsset, amount, limitOrders }: SwapRateArgs) => {
+const getDeductedBrokerFeeOutput = ({
+  destAsset,
+  inputAmount,
+  intermediateAmount,
+  outputAmount,
+  brokerCommissionBps,
+  egressFee,
+}: {
+  destAsset: InternalAsset;
+  inputAmount: bigint;
+  intermediateAmount: bigint | null;
+  outputAmount: bigint;
+  brokerCommissionBps?: number;
+  egressFee: bigint;
+}) => {
+  if (!brokerCommissionBps) {
+    return { intermediateAmount, outputAmount };
+  }
+  let usdcAmount = intermediateAmount ?? inputAmount;
+  if (destAsset === 'Usdc') {
+    usdcAmount = outputAmount;
+  }
+
+  const brokerFee = BigInt(
+    brokerCommissionBps && getPipAmountFromAmount(usdcAmount, brokerCommissionBps),
+  );
+
```

Comment:
i would remove the broker fee up here because any following calculation will have to do it otherwise